warehouse/query-core/src/main/java/datawave/query/jexl/visitors/IsNotNullPruningVisitor.java

```diff
@@ -121,34 +116,36 @@ public static JexlNode prune(JexlNode node) {
      */
     @Override
     public Object visit(ASTAndNode node, Object data) {
-        
-        boolean notNullExists = false;
-        Set<String> fields = new HashSet<>(node.jjtGetNumChildren());
-        for (JexlNode child : JexlNodes.children(node)) {
-            notNullExists |= findEqualityFieldsForNode(child, fields);
-        }
-        
-        // only rebuild if it's possible
-        if (notNullExists && !fields.isEmpty()) {
-            List<JexlNode> next = new ArrayList<>(node.jjtGetNumChildren());
-            JexlNode built;
-            for (JexlNode child : JexlNodes.children(node)) {
-                built = pruneNode(child, fields);
-                if (built != null) {
-                    next.add(built);
-                }
+        // do not process marker nodes
+        if (!QueryPropertyMarkerVisitor.getInstance(node).isAnyType()) {
```

Comment:
This is a good functional change

Reply:
You'll notice this in a lot of different visitors, and that's because these checks used to be done for ASTReference.  Now that that node is no longer created, the checks had to move down to the next node in the marker topology - the and node.  Just providing some additional context...